File: monitorS/client_tray.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
员工监控系统 - 托盘图标模块（多语言版）- 修复版
"""

import os
import sys
import time
import threading
import logging
import webbrowser
from pathlib import Path

# 尝试导入托盘图标依赖
try:
    from PIL import Image, ImageDraw
    import pystray

    PYSTRAY_AVAILABLE = True
except ImportError:
    PYSTRAY_AVAILABLE = False

try:
    from plyer import notification

    PLYER_AVAILABLE = True
except ImportError:
    PLYER_AVAILABLE = False

try:
    import winreg

    WINREG_AVAILABLE = True
except ImportError:
    WINREG_AVAILABLE = False

from client_i18n import get_text


class EnhancedTrayIcon:
    """增强的托盘图标类（多语言版）"""

    def __init__(self, client):
        self.client = client
        self.logger = logging.getLogger(__name__)

        # 通知冷却
        self.last_notification = {}
        self.notification_cooldown = 60

        # 依赖检查
        self.pystray_available = PYSTRAY_AVAILABLE
        self.plyer_available = PLYER_AVAILABLE

        self.icon = None

        if self.pystray_available:
            self.create_icon()
        else:
            self.logger.warning("⚠️ pystray未安装，托盘图标不可用")

    def create_icon(self):
        """创建托盘图标"""
        if not self.pystray_available:
            return

        # 创建图标图像
        image = self._create_icon_image()

        # 创建菜单
        menu = (
            pystray.MenuItem(get_text("show_status"), self.show_status),
            pystray.MenuItem(get_text("health_status"), self.show_health_status),
            pystray.MenuItem(get_text("watchdog_status"), self.show_watchdog_status),
            pystray.Menu.SEPARATOR,
            pystray.MenuItem(
                get_text("pause_monitor"),
                self.pause_monitor,
                enabled=lambda item: not self.client.paused,
            ),
            pystray.MenuItem(
                get_text("resume_monitor"),
                self.resume_monitor,
                enabled=lambda item: self.client.paused,
            ),
            pystray.MenuItem(get_text("screenshot_now"), self.take_screenshot_now),
            pystray.Menu.SEPARATOR,
            pystray.MenuItem(get_text("upload_queue"), self.show_queue_status),
            pystray.MenuItem(get_text("cleanup_cache"), self.cleanup_cache),
            pystray.MenuItem(get_text("network_diagnostic"), self.network_diagnostic),
            pystray.Menu.SEPARATOR,
            pystray.MenuItem(get_text("reconfigure"), self.reconfigure),
            pystray.MenuItem(
                get_text("autostart"),
                self.toggle_autostart,
                checked=lambda item: self.is_autostart_enabled(),
            ),
            pystray.MenuItem(get_text("view_log"), self.open_log),
            pystray.Menu.SEPARATOR,
            pystray.MenuItem(get_text("exit"), self.exit_app),
        )

        self.icon = pystray.Icon("employee_monitor", image, "员工监控系统", menu)

    def _create_icon_image(self):
        """创建图标图像"""
        image = Image.new("RGB", (64, 64), color=(102, 126, 234))
        draw = ImageDraw.Draw(image)

        # 绘制相机图标
        draw.rectangle([16, 20, 48, 44], outline=(255, 255, 255), width=2)
        draw.ellipse([28, 28, 36, 36], fill=(255, 255, 255))
        draw.line([40, 28, 44, 24], fill=(255, 255, 255), width=2)

        # 状态指示
        if self.client.paused:
            draw.ellipse([48, 8, 56, 16], fill=(255, 255, 0), outline=(0, 0, 0))
        elif self.client.offline_mode:
            draw.ellipse([48, 8, 56, 16], fill=(255, 0, 0), outline=(0, 0, 0))
        else:
            draw.ellipse([48, 8, 56, 16], fill=(0, 255, 0), outline=(0, 0, 0))

        return image

    def show_status(self):
        """显示基本状态"""
        stats = self.client.get_stats()
        uptime = stats.get("uptime", 0)
        hours = int(uptime // 3600)
        minutes = int((uptime % 3600) // 60)

        # 修复1：使用硬编码文本，避免缺少国际化键
        message = (
            f"员工ID: {self.client.employee_id}\n"
            f"服务器: {self.client.current_server}\n"
            f"运行时间: {hours}小时{minutes}分钟\n"
            f"截图数量: {stats['screenshots_taken']}\n"
            f"上传成功: {stats['screenshots_uploaded']}\n"
            f"跳过相似: {stats.get('skipped_similar', 0)}\n"
            f"截图间隔: {self.client.interval}秒"
        )

        self._show_notification(get_text("show_status"), message)

    def show_health_status(self):
        """显示健康状态"""
        if not hasattr(self.client, "health_monitor"):
            self._show_notification(get_text("health_status"), "健康监控未启用")
            return

        summary = self.client.health_monitor.get_summary()

        message = (
            f"{get_text('health_status')}:\n"
            f"健康: {summary['healthy']}\n"
            f"降级: {summary['degraded']}\n"
            f"异常: {summary['unhealthy']}\n"
            f"未知: {summary['unknown']}\n"
            f"健康率: {summary['health_rate']*100:.1f}%\n"
            f"恢复次数: {summary['total_recoveries']}"
        )

        self._show_notification(get_text("health_status"), message)

    def show_watchdog_status(self):
        """显示看门狗状态"""
        if not hasattr(self.client, "watchdog"):
            self._show_notification(get_text("watchdog_status"), "看门狗未启用")
            return

        status = self.client.watchdog.get_status()

        # 修复2：使用硬编码文本
        running_text = "运行中" if status["running"] else "已停止"

        message = (
            f"{get_text('watchdog_status')}:\n"
            f"{running_text}\n"
            f"总重启: {status['stats']['total_restarts']}\n"
            f"失败重启: {status['stats']['failed_restarts']}\n\n"
            f"监控进程:\n"
        )

        for name, proc in status["processes"].items():
            icon = "✅" if proc["alive"] else "❌"
            message += f"{icon} {name}: {proc['status']}"
            if proc["restart_count"] > 0:
                message += f" (重启:{proc['restart_count']})"
            message += "\n"

        self._show_notification(get_text("watchdog_status"), message)

    def show_queue_status(self):
        """显示队列状态"""
        if not hasattr(self.client, "upload_queue"):
            self._show_notification(get_text("upload_queue"), "上传队列未启用")
            return

        stats = self.client.upload_queue.get_stats()
        queue_percent = (stats["queue_size"] / stats["queue_maxsize"]) * 100

        message = (
            f"队列大小: {stats['queue_size']}/{stats['queue_maxsize']} ({queue_percent:.1f}%)\n"
            f"已处理: {stats['processed']}\n"
            f"失败: {stats['failed']}\n"
            f"重试: {stats['retried']}\n"
            f"缓存文件: {stats['cache_count']}\n"
            f"缓存大小: {stats['cache_size']/1024/1024:.1f}MB"
        )

        self._show_notification(get_text("upload_queue"), message)

    def pause_monitor(self):
        """暂停监控"""
        self.client.paused = True
        self._show_notification(
            get_text("notification_monitor"), get_text("msg_paused")
        )
        self.update_icon_title()

    def resume_monitor(self):
        """恢复监控"""
        self.client.paused = False
        self._show_notification(
            get_text("notification_monitor"), get_text("msg_resumed")
        )
        self.update_icon_title()

    def take_screenshot_now(self):
        """立即截图"""
        self.client.take_screenshot_now = True
        self._show_notification(
            get_text("notification_screenshot"), get_text("msg_screenshot_triggered")
        )

    def cleanup_cache(self):
        """清理缓存"""
        if hasattr(self.client, "upload_queue"):
            self.client.upload_queue._cleanup_cache()
            self._show_notification(
                get_text("notification_cache"), get_text("msg_cache_cleaned")
            )

    def network_diagnostic(self):
        """网络诊断"""
        # 修复3：使用硬编码文本
        mode_text = "在线" if not self.client.offline_mode else "离线"
        heartbeat_text = "已启用" if self.client.enable_heartbeat else "已禁用"

        message = (
            f"当前服务器: {self.client.current_server}\n"
            f"Client ID: {self.client.client_id}\n"
            f"模式: {mode_text}\n"
            f"心跳: {heartbeat_text}\n\n"
        )

        # 测试连接
        try:
            import requests

            start = time.time()
            response = requests.get(
                f"{self.client.current_server}/health", timeout=5, verify=False
            )
            elapsed = (time.time() - start) * 1000

            if response.status_code == 200:
                message += f"✅ 服务器连接正常 (延迟: {elapsed:.0f}ms)"
            else:
                message += f"⚠️ 服务器状态码: {response.status_code}"
        except Exception as e:
            message += f"❌ 连接失败: {e}"

        self._show_notification(get_text("network_diagnostic"), message)

    def reconfigure(self):
        """重新配置"""
        import tkinter as tk
        from tkinter import messagebox

        root = tk.Tk()
        root.withdraw()

        result = messagebox.askyesno(
            get_text("reconfigure_title"),
            get_text("reconfigure_message"),
            parent=root,
        )

        if result:
            self.logger.info("✏️ 用户请求重新配置")
            self.client.config_manager.set("employee_name", "")
            self.client.config_manager.set("client_id", "")
            self.client.config_manager.set("employee_id", "")
            self.client._force_reconfigure = True
            self.client.first_run = True

            success = self.client.register_with_server(silent_mode=False)

            if success:
                self._show_notification(
                    get_text("reconfigure_success"), get_text("reconfigure_success_msg")
                )
            else:
                self._show_notification(
                    get_text("reconfigure_failed"), get_text("reconfigure_failed_msg")
                )

        root.destroy()

    def toggle_autostart(self):
        """切换开机自启"""
        if self.is_autostart_enabled():
            self.disable_autostart()
        else:
            self.enable_autostart()

    def is_autostart_enabled(self) -> bool:
        """检查开机自启"""
        if not WINREG_AVAILABLE or sys.platform != "win32":
            return False

        try:
            key = winreg.OpenKey(
                winreg.HKEY_CURRENT_USER,
                r"Software\Microsoft\Windows\CurrentVersion\Run",
                0,
                winreg.KEY_READ,
            )
            winreg.QueryValueEx(key, "EmployeeMonitor")
            winreg.CloseKey(key)
            return True
        except:
            return False

    def enable_autostart(self):
        """启用开机自启"""
        if not WINREG_AVAILABLE or sys.platform != "win32":
            return

        try:
            key = winreg.OpenKey(
                winreg.HKEY_CURRENT_USER,
                r"Software\Microsoft\Windows\CurrentVersion\Run",
                0,
                winreg.KEY_SET_VALUE,
            )

            if getattr(sys, "frozen", False):
                exe_path = sys.executable
            else:
                exe_path = f'"{sys.executable}" "{__file__}"'

            winreg.SetValueEx(key, "EmployeeMonitor", 0, winreg.REG_SZ, exe_path)
            winreg.CloseKey(key)
        except Exception as e:
            self.logger.error(f"设置开机自启失败: {e}")

    def disable_autostart(self):
        """禁用开机自启"""
        if not WINREG_AVAILABLE or sys.platform != "win32":
            return

        try:
            key = winreg.OpenKey(
                winreg.HKEY_CURRENT_USER,
                r"Software\Microsoft\Windows\CurrentVersion\Run",
                0,
                winreg.KEY_SET_VALUE,
            )
            winreg.DeleteValue(key, "EmployeeMonitor")
            winreg.CloseKey(key)
        except:
            pass

    def open_log(self):
        """打开日志"""
        log_file = "monitor.log"
        if os.path.exists(log_file):
            if sys.platform == "win32":
                os.startfile(log_file)
            elif sys.platform == "darwin":
                import subprocess

                subprocess.run(["open", log_file])

    def exit_app(self):
        """退出程序"""
        self._show_notification(get_text("notification_exit"), get_text("msg_exiting"))
        self.client.stop()
        if self.icon:
            self.icon.stop()

    def _show_notification(self, title: str, message: str):
        """显示通知"""
        now = time.time()
        key = f"{title}_{message[:50]}"

        if key in self.last_notification:
            if now - self.last_notification[key] < self.notification_cooldown:
                return

        self.last_notification[key] = now

        if self.plyer_available:
            try:
                notification.notify(
                    title=f"员工监控系统 - {title}", message=message, timeout=5
                )
            except Exception as e:
                self.logger.error(f"通知失败: {e}")
                print(f"{title}: {message}")
        else:
            print(f"{title}: {message}")

    def update_icon_title(self):
        """更新图标标题"""
        if self.icon:
            if self.client.paused:
                self.icon.title = "员工监控系统 - 已暂停"
            elif self.client.offline_mode:
                self.icon.title = "员工监控系统 - 离线模式"
            else:
                stats = self.client.get_stats()
                self.icon.title = (
                    f"员工监控系统 - {stats.get('screenshots_taken', 0)}张截图"
                )

    def run(self):
        """运行托盘图标"""
        if self.icon:
            self.icon.run()

File: monitor-go/client_tray.go
// client_tray.go - 完善菜单处理

package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
	"sync"
	"sync/atomic"
	"time"
)

// TrayIcon 托盘图标管理器
type TrayIcon struct {
    client               *MonitorClient
    running              bool
    lastNotification     map[string]time.Time
    notificationCooldown time.Duration
    mu                   sync.RWMutex
    reader               *bufio.Reader
}

// NewTrayIcon 创建托盘图标
func NewTrayIcon(client *MonitorClient) *TrayIcon {
    return &TrayIcon{
        client:               client,
        lastNotification:     make(map[string]time.Time),
        notificationCooldown: 60 * time.Second,
        reader:               bufio.NewReader(os.Stdin),
    }
}

// Run 运行托盘图标
func (t *TrayIcon) Run() {
    t.running = true
    t.showConsoleMenu()
}

// Stop 停止托盘图标
func (t *TrayIcon) Stop() {
    t.running = false
}

func (t *TrayIcon) showConsoleMenu() {
    for t.running {
        fmt.Println("\n" + strings.Repeat("=", 50))
        fmt.Println("          员工监控系统 v" + Version)
        fmt.Println(strings.Repeat("=", 50))
        
        // 显示当前状态
        t.showStatusLine()
        
        fmt.Println("\n请选择操作:")
        fmt.Println(" 1. 显示详细状态")
        fmt.Println(" 2. 健康状态")
        fmt.Println(" 3. 看门狗状态")
        fmt.Println(" 4. 上传队列状态")
        fmt.Println(" 5. 远程屏幕状态")
        fmt.Println(" 6. 暂停监控")
        fmt.Println(" 7. 恢复监控")
        fmt.Println(" 8. 立即截图")
        fmt.Println(" 9. 清理缓存")
        fmt.Println("10. 网络诊断")
        fmt.Println("11. 重新配置")
        fmt.Println("12. 查看日志")
        
        // ========== 新增菜单项 ==========
        fmt.Println("13. 硬件信息")
        
        autoStartStatus := "❌ 未启用"
        if t.client.IsAutoStartEnabled() {
            autoStartStatus = "✅ 已启用"
        }
        fmt.Printf("14. 开机自启 [%s]\n", autoStartStatus)
        // ================================
        
        fmt.Println("15. 退出程序")
        fmt.Print("\n请输入选项 (1-15): ")
        
        input, err := t.reader.ReadString('\n')
        if err != nil {
            continue
        }
        input = strings.TrimSpace(input)
        
        switch input {
        case "1":
            t.ShowStatus()
        case "2":
            t.ShowHealthStatus()
        case "3":
            t.ShowWatchdogStatus()
        case "4":
            t.ShowQueueStatus()
        case "5":
            t.ShowRemoteStatus()
        case "6":
            t.PauseMonitor()
        case "7":
            t.ResumeMonitor()
        case "8":
            t.TakeScreenshotNow()
        case "9":
            t.CleanupCache()
        case "10":
            t.NetworkDiagnostic()
        case "11":
            t.Reconfigure()
        case "12":
            t.ViewLog()
        // ========== 新增 case ==========
        case "13":
            t.ShowHardwareInfo()
        case "14":
            t.ToggleAutoStartMenu()
        // ================================
        case "15":
            t.ExitApp()
        default:
            fmt.Println("无效选项，请重新输入")
        }
        
        time.Sleep(500 * time.Millisecond)
    }
}

func (t *TrayIcon) showStatusLine() {
    if t.client == nil {
        return
    }
    
    statusIcon := "🟢"
    if atomic.LoadInt32(&t.client.paused) == 1 {
        statusIcon = "⏸️"
    } else if atomic.LoadInt32(&t.client.offlineMode) == 1 {
        statusIcon = "🔴"
    }
    
    stats := t.client.GetStats()
    screenshots, _ := stats["screenshots_taken"].(int)
    
    fmt.Printf("\n状态: %s | 截图: %d | 员工: %s",
        statusIcon, screenshots, t.client.employeeID)
}

// ShowRemoteStatus 显示远程屏幕状态
func (t *TrayIcon) ShowRemoteStatus() {
    if t.client == nil || t.client.remoteScreen == nil {
        t.showNotification("远程屏幕", "远程屏幕未启用")
        return
    }
    
    status := t.client.remoteScreen.GetStatus()
    
    message := fmt.Sprintf(
        "远程屏幕状态:\n"+
            "状态: %s\n"+
            "观众数: %d\n"+
            "当前帧率: %d fps\n"+
            "当前画质: %d%%\n"+
            "当前宽度: %d\n"+
            "已发送帧: %d\n"+
            "已发送字节: %.2f MB\n"+
            "丢弃帧: %d\n"+
            "网络质量: %d",
        status.State,
        status.ViewerCount,
        status.CurrentFPS,
        status.CurrentQuality,
        status.CurrentWidth,
        status.FramesSent,
        float64(status.BytesSent)/1024/1024,
        status.DroppedFrames,
        status.NetworkQuality,
    )
    
    t.showNotification("远程屏幕状态", message)
}

// ViewLog 查看日志
func (t *TrayIcon) ViewLog() {
    logFile := "logs/monitor.log"
    data, err := os.ReadFile(logFile)
    if err != nil {
        t.showNotification("查看日志", "无法读取日志文件: "+err.Error())
        return
    }
    
    // 只显示最后30行
    lines := strings.Split(string(data), "\n")
    start := len(lines) - 30
    if start < 0 {
        start = 0
    }
    
    fmt.Println("\n" + strings.Repeat("=", 60))
    fmt.Println("最近日志 (最后30行)")
    fmt.Println(strings.Repeat("=", 60))
    
    for i := start; i < len(lines); i++ {
        if lines[i] != "" {
            fmt.Println(lines[i])
        }
    }
    
    fmt.Println(strings.Repeat("=", 60))
}

// ShowStatus 显示状态
func (t *TrayIcon) ShowStatus() {
    if t.client == nil {
        return
    }
    
    stats := t.client.GetStats()
    uptime, _ := stats["uptime"].(float64)
    hours := int(uptime / 3600)
    minutes := int((uptime - float64(hours*3600)) / 60)
    
    // 获取队列统计
    queueStats := ""
    if t.client.uploadQueue != nil {
        qs := t.client.uploadQueue.GetStats()
        queueStats = fmt.Sprintf("\n队列: %d/%d", qs["queue_size"], qs["queue_maxsize"])
    }
    
    message := fmt.Sprintf(
        "员工ID: %s\n"+
            "客户端ID: %s\n"+
            "服务器: %s\n"+
            "运行时间: %d小时%d分钟\n"+
            "截图数量: %d\n"+
            "上传成功: %d\n"+
            "跳过相似: %d\n"+
            "上传失败: %d\n"+
            "截图间隔: %.0f秒\n"+
            "离线模式: %v\n"+
            "暂停状态: %v%s",
        t.client.employeeID,
        t.client.clientID,
        t.client.currentServer,
        hours,
        minutes,
        stats["screenshots_taken"],
        stats["screenshots_uploaded"],
        stats["skipped_similar"],
        stats["upload_failures"],
        t.client.interval.Seconds(),
        t.client.offlineMode == 1,
        t.client.paused == 1,
        queueStats,
    )
    
    t.showNotification("系统状态", message)
}

// ShowHealthStatus 显示健康状态
func (t *TrayIcon) ShowHealthStatus() {
    if t.client == nil || t.client.healthMonitor == nil {
        t.showNotification("健康状态", "健康监控未启用")
        return
    }

    summary := t.client.healthMonitor.GetSummary()
    message := fmt.Sprintf(
        "健康状态:\n"+
            "健康: %d\n"+
            "降级: %d\n"+
            "异常: %d\n"+
            "未知: %d\n"+
            "健康率: %.1f%%\n"+
            "恢复次数: %d",
        summary["healthy"],
        summary["degraded"],
        summary["unhealthy"],
        summary["unknown"],
        summary["health_rate"].(float64)*100,
        summary["total_recoveries"],
    )

    t.showNotification("健康状态", message)
}

// ShowWatchdogStatus 显示看门狗状态
func (t *TrayIcon) ShowWatchdogStatus() {
    if t.client == nil || t.client.watchdog == nil {
        t.showNotification("看门狗状态", "看门狗未启用")
        return
    }

    status := t.client.watchdog.GetStatus()
    runningText := "运行中"
    if !status["running"].(bool) {
        runningText = "已停止"
    }

    stats := status["stats"].(map[string]interface{})
    message := fmt.Sprintf(
        "看门狗状态:\n"+
            "%s\n"+
            "总重启: %d\n"+
            "失败重启: %d\n\n"+
            "监控进程:\n",
        runningText,
        stats["total_restarts"],
        stats["failed_restarts"],
    )

    processes := status["processes"].(map[string]interface{})
    for name, proc := range processes {
        procMap := proc.(map[string]interface{})
        icon := "✅"
        if !procMap["alive"].(bool) {
            icon = "❌"
        }
        message += fmt.Sprintf("%s %s: %s", icon, name, procMap["status"])
        if procMap["restart_count"].(int) > 0 {
            message += fmt.Sprintf(" (重启:%d)", procMap["restart_count"])
        }
        message += "\n"
    }

    t.showNotification("看门狗状态", message)
}

// ShowQueueStatus 显示队列状态
func (t *TrayIcon) ShowQueueStatus() {
    if t.client == nil || t.client.uploadQueue == nil {
        t.showNotification("上传队列", "上传队列未启用")
        return
    }

    stats := t.client.uploadQueue.GetStats()
    queuePercent := float64(stats["queue_size"].(int)) / float64(stats["queue_maxsize"].(int)) * 100
    cacheSizeMB := float64(stats["cache_size"].(int64)) / 1024 / 1024

    message := fmt.Sprintf(
        "上传队列状态:\n"+
            "队列大小: %d/%d (%.1f%%)\n"+
            "已处理: %d\n"+
            "失败: %d\n"+
            "重试: %d\n"+
            "缓存文件: %d\n"+
            "缓存大小: %.2f MB\n"+
            "背压计数: %d\n"+
            "运行状态: %v",
        stats["queue_size"],
        stats["queue_maxsize"],
        queuePercent,
        stats["processed"],
        stats["failed"],
        stats["retried"],
        stats["cache_count"],
        cacheSizeMB,
        stats["backpressure"],
        stats["is_running"],
    )

    t.showNotification("上传队列", message)
}

// PauseMonitor 暂停监控
func (t *TrayIcon) PauseMonitor() {
    if t.client == nil {
        return
    }
    atomic.StoreInt32(&t.client.paused, 1)
    t.showNotification("监控", "监控已暂停")
}

// ResumeMonitor 恢复监控
func (t *TrayIcon) ResumeMonitor() {
    if t.client == nil {
        return
    }
    atomic.StoreInt32(&t.client.paused, 0)
    t.showNotification("监控", "监控已恢复")
}

// TakeScreenshotNow 立即截图
func (t *TrayIcon) TakeScreenshotNow() {
    if t.client == nil {
        return
    }
    atomic.StoreInt32(&t.client.takeScreenshotNow, 1)
    t.showNotification("截图", "已触发立即截图")
}

// CleanupCache 清理缓存
func (t *TrayIcon) CleanupCache() {
    if t.client != nil && t.client.uploadQueue != nil {
        t.client.uploadQueue.cleanupCache()
        t.showNotification("缓存清理", "缓存清理完成")
    }
}

// NetworkDiagnostic 网络诊断
func (t *TrayIcon) NetworkDiagnostic() {
    if t.client == nil {
        return
    }
    
    modeText := "在线"
    if t.client.offlineMode == 1 {
        modeText = "离线"
    }
    heartbeatText := "已启用"
    if !t.client.enableHeartbeat {
        heartbeatText = "已禁用"
    }
    
    // 测试网络连接
    networkOK := t.client.checkBasicNetwork()
    serverOK := false
    if networkOK && t.client.apiClient != nil {
        _, err := t.client.apiClient.Get("/health")
        serverOK = err == nil
    }
    
    message := fmt.Sprintf(
        "网络诊断结果:\n"+
            "当前服务器: %s\n"+
            "Client ID: %s\n"+
            "员工ID: %s\n"+
            "模式: %s\n"+
            "心跳: %s\n"+
            "基础网络: %v\n"+
            "服务器连接: %v",
        t.client.currentServer,
        t.client.clientID,
        t.client.employeeID,
        modeText,
        heartbeatText,
        map[bool]string{true: "✅ 正常", false: "❌ 异常"}[networkOK],
        map[bool]string{true: "✅ 正常", false: "❌ 异常"}[serverOK],
    )
    
    t.showNotification("网络诊断", message)
}

// Reconfigure 重新配置
func (t *TrayIcon) Reconfigure() {
    if t.client == nil {
        return
    }
    
    fmt.Println("\n" + strings.Repeat("=", 50))
    fmt.Println("重新配置模式")
    fmt.Println("程序将在退出后需要重新配置")
    fmt.Println(strings.Repeat("=", 50))
    
    fmt.Print("确认重新配置? (y/n): ")
    input, _ := t.reader.ReadString('\n')
    input = strings.TrimSpace(strings.ToLower(input))
    
    if input == "y" || input == "yes" {
        t.client.forceReconfigure = true
        t.client.firstRun = true
        
        // 保存配置标记
        if t.client.configManager != nil {
            t.client.configManager.Set("force_reconfigure", true)
        }
        
        t.showNotification("重新配置", "请在重启后重新配置")
    }
}

// ExitApp 退出程序
func (t *TrayIcon) ExitApp() {
    fmt.Println("\n正在退出程序...")
    t.showNotification("退出", "正在退出程序...")
    if t.client != nil {
        t.client.Stop()
    }
    t.running = false
}

func (t *TrayIcon) showNotification(title, message string) {
    t.mu.Lock()
    defer t.mu.Unlock()
    
    now := time.Now()
    key := title + "_" + message[:minInt(len(message), 50)]

    if lastTime, ok := t.lastNotification[key]; ok {
        if now.Sub(lastTime) < t.notificationCooldown {
            return
        }
    }

    t.lastNotification[key] = now

    fmt.Printf("\n[%s] %s:\n%s\n", time.Now().Format("15:04:05"), title, message)
}

func minInt(a, b int) int {
    if a < b {
        return a
    }
    return b
}

// ShowHardwareInfo 显示硬件信息
func (t *TrayIcon) ShowHardwareInfo() {
    if t.client == nil {
        return
    }
    
    info := t.client.GetHardwareInfo()
    
    // 构建显示信息
    message := fmt.Sprintf(
        "========== 硬件信息 ==========\n"+
            "CPU: %s\n"+
            "CPU ID: %s\n"+
            "CPU核心: %d\n"+
            "\n主板: %s\n"+
            "主板序列号: %s\n"+
            "\n磁盘: %s\n"+
            "磁盘序列号: %s\n"+
            "\n系统UUID: %s\n"+
            "BIOS: %s\n"+
            "\n硬件指纹: %s\n"+
            "指纹可信度: %.1f%%\n"+
            "=============================",
        truncateStringForTray(info.CPUName, 50),
        info.CPUID,
        info.CPUCores,
        info.MotherboardModel,
        info.MotherboardSN,
        info.DiskModel,
        info.DiskSerial,
        info.UUID,
        info.BIOSVersion,
        info.Fingerprint[:16]+"...",
        info.Reliability*100,
    )
    
    t.showNotification("硬件信息", message)
}

// ToggleAutoStartMenu 切换开机自启（带交互）
func (t *TrayIcon) ToggleAutoStartMenu() {
    enabled := t.client.IsAutoStartEnabled()
    
    if enabled {
        fmt.Print("\n确认禁用开机自启? (y/n): ")
        response, _ := t.reader.ReadString('\n')
        if strings.TrimSpace(strings.ToLower(response)) != "y" {
            t.showNotification("开机自启", "操作已取消")
            return
        }
        
        if err := t.client.DisableAutoStart(); err != nil {
            t.showNotification("开机自启", "禁用失败: "+err.Error())
        } else {
            t.showNotification("开机自启", "已禁用开机自启")
        }
    } else {
        fmt.Print("\n确认启用开机自启? (y/n): ")
        response, _ := t.reader.ReadString('\n')
        if strings.TrimSpace(strings.ToLower(response)) != "y" {
            t.showNotification("开机自启", "操作已取消")
            return
        }
        
        if err := t.client.EnableAutoStart(); err != nil {
            t.showNotification("开机自启", "启用失败: "+err.Error())
        } else {
            t.showNotification("开机自启", "已启用开机自启")
            
            // 询问是否设置延迟启动
            fmt.Print("\n是否设置延迟启动? (y/n): ")
            delayResp, _ := t.reader.ReadString('\n')
            if strings.TrimSpace(strings.ToLower(delayResp)) == "y" {
                fmt.Print("延迟秒数 (默认10): ")
                delayStr, _ := t.reader.ReadString('\n')
                delay := 10
                if d, err := strconv.Atoi(strings.TrimSpace(delayStr)); err == nil && d > 0 {
                    delay = d
                }
                if err := t.client.SetAutoStartDelay(delay); err == nil {
                    t.showNotification("开机自启", fmt.Sprintf("已设置延迟启动 %d 秒", delay))
                }
            }
        }
    }
}

// truncateStringForTray 截断字符串用于托盘显示
func truncateStringForTray(s string, maxLen int) string {
    if len(s) <= maxLen {
        return s
    }
    return s[:maxLen-3] + "..."
}